src/DocumentFormat.OpenXml.Linq/OpenXmlLinqExtensions.cs

```diff
@@ -0,0 +1,169 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Framework.Features;
+using DocumentFormat.OpenXml.Packaging;
+using System;
+using System.IO;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace DocumentFormat.OpenXml.Linq
+{
+    /// <summary>
+    /// Extensions to access an <see cref="XElement"/> instance for parts.
+    /// </summary>
+    public static class OpenXmlLinqExtensions
+    {
+        /// <summary>
+        /// Will create or get a cached instance of <see cref="IPartRootXElementFeature"/>.
+        /// </summary>
+        /// <param name="part">The part to provide an <see cref="XElement"/> instance.</param>
+        /// <returns>A <see cref="IPartRootXElementFeature"/>.</returns>
+        public static IPartRootXElementFeature GetPartRootXElementFeature(this OpenXmlPart part)
+        {
+            var feature = part.Features.Get<IPartRootXElementFeature>();
+
+            if (feature is not null)
+            {
+                return feature;
+            }
+
+            part.TryAddDisposableFeature();
+            part.TryAddPartRootEventsFeature();
+
+            var xelement = new RootXElementFeature(part);
+
+            part.Features.SetDisposable<IPartRootXElementFeature>(xelement);
+
+            return xelement;
+        }
+
+        /// <summary>
+        /// Gets an <see cref="XElement"/> representation of the <paramref name="part"/>.
+        /// </summary>
+        /// <param name="part">The part to get the contents of.</param>
+        /// <returns>An <see cref="XElement"/>.</returns>
+        public static XElement? GetXElement(this OpenXmlPart part)
+            => part.GetPartRootXElementFeature().Root;
+
+        /// <summary>
+        /// Sets an <see cref="XElement"/> to the part. This will force a save to the underlying stream.
+        /// </summary>
+        /// <param name="part">The part to set to.</param>
+        /// <param name="xElement">The element to set.</param>
+        public static void SetXElement(this OpenXmlPart part, XElement? xElement)
+            => part.GetPartRootXElementFeature().Root = xElement;
+
+        /// <summary>
+        /// Saves the current <see cref="XElement"/> to the part.
+        /// </summary>
+        /// <param name="part">The part to save to.</param>
+        public static void SaveXElement(this OpenXmlPart part)
+            => part.GetPartRootXElementFeature().Save();
+
+        private sealed class RootXElementFeature : IPartRootXElementFeature, IDisposable
+        {
+            private readonly OpenXmlPart _part;
+            private readonly IPartRootEventsFeature _events;
+
+            private XElement? _rootXElement;
+
+            public RootXElementFeature(OpenXmlPart part)
+            {
+                _part = part;
+                _events = _part.Features.GetRequired<IPartRootEventsFeature>();
+
+                _events.Change += OnPartRootChange;
```

Comment:
@twsouthwick, I'm not suggesting a change here but wanted to note that only `Saving` and `Saved` events are raised by the `OpenXmlPartRootElement`. However, I'm thinking we might also want to have `OpenXmlPartRootElement` raise the `Reloading` and `Reloaded` events that you have already defined. For example, consider the case where the `_rootXElement` was deserialized from an `OpenXmlPartRootElement` instance that was changed after loading it from the `OpenXmlPart`. If the user discards those changes by calling the `Reload()` method, the `_rootXElement` should also be discarded and loaded again once it is accessed.

Reply:
I've created an issue to track that for a future change.